packages/core/src/extension/extension-event-broadcaster.ts

```diff
@@ -0,0 +1,94 @@
+import { ReadableCheckoutStore } from '../checkout';
+import { IframeEventListener } from '../common/iframe';
+
+import { ExtensionEvent, ExtensionEventType } from './extension-client';
+import {
+    ExtensionInternalCommandMap,
+    ExtensionInternalCommandType,
+    ExtensionSubscribeCommand,
+    ExtensionUnsubscribeCommand,
+} from './extension-internal-commands';
+import { ExtensionMessenger } from './extension-messenger';
+import { ExtensionChangeSubscriber, ExtensionChangeUnsubscriber } from './subscribers';
+
+export class ExtensionEventBroadcaster {
+    private _subscribed: { [id: string]: boolean } = {};
+    private _unsubscribers: { [type: string]: ExtensionChangeUnsubscriber } = {};
+    private _listeners: { [id: string]: IframeEventListener<ExtensionInternalCommandMap> } = {};
+
+    constructor(
+        private _store: ReadableCheckoutStore,
+        private _extensionMessenger: ExtensionMessenger,
+        private _subscribers: Record<ExtensionEventType, ExtensionChangeSubscriber>,
+    ) {}
+
+    listen(): void {
```

Comment:
Just wondering what will be the use case for this 🤔

Reply:
It's called when the first extension is rendered, see https://github.com/bigcommerce/checkout-sdk-js/blob/dcbdcaa5d48947d5dd6c0460837eae87eddcd237/packages/core/src/checkout/checkout-service.ts#L1405-L1412